### PR TITLE
KRP-1229 - Legal Representation creation

### PR DESCRIPTION
**Pull request by jablonskipj**
[KRP-1229](https://linear.app/ageras-group/issue/KRP-1229/[mocksolaris]-create-legal-representative-on-mock-solaris)

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -892,6 +892,12 @@ router.post(
   safeRequestHandler(businessesAPI.createBeneficialOwner)
 );
 
+router.post(
+  "/businesses/:business_id/legal_representative",
+  middlewares.withPerson,
```
Comment (lera) [inline src/app.ts:897]:
```suggestion
  middlewares.withBusiness,
```

File: src/routes/business/legalRepresentative.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  LegalRepresentative,
+  LegalRepresentativeType,
+  LegalRepresentativeRepresentationType,
+} from "../../helpers/types";
+
+export const createLegalRepresentative = async (
+  req: Request,
+  res: Response
+) => {
+  const { business_id: businessId } = req.params;
```
Comment (lera) [inline src/routes/business/legalRepresentative.ts]: by using `withBusiness` middleware, you can get business entity directly from request

File: src/routes/business/legalRepresentative.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      business.legalRepresentatives.push({ ...legalRepresentative });
+    }
+
+    await saveBusiness(business);
+
+    return res.status(201).send(legalRepresentative);
+  } catch (err) {
+    if (
+      err.message ===
+      `Business which has businessId: ${businessId} was not found in redis`
```
Comment (lera) [inline src/routes/business/legalRepresentative.ts]: this error will be also thrown from the middleware

File: src/app.ts (hunk 1 above)
Comment (mbayraktr) [reply src/app.ts:897]: But don't we need person middleware as well ? I mean what if person is not there yet 🤔

Comment (lera) [reply src/app.ts:897]: you provide person_id in the payload, it already means that the person exists and there is no need for fetching this entity

Comment (lera) [reply src/app.ts:897]: @jablonskipj there is even no person_id in params, please remove it

File: tests/routes/business/legalRepresentative.spec.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      res = mockRes();
+      req = mockReq({
+        params: {
+          business_id: businessId,
+        },
+        body: {
+          person_id: "1234abcdef",
+          voting_share: 0.5,
+          fictitious: false,
+          relationship_to_business: "owner",
```
Comment (lera) [inline tests/routes/business/legalRepresentative.spec.ts]: too much copy pasta from BO

File: src/routes/business/legalRepresentative.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  const { business } = req;
+
+  try {
+    const legalRepresentative: LegalRepresentative = {
+      id: generateID(),
+      legal_representative_id: req.body.person_id,
+      business_id: business.id,
+      valid_until: null,
+      legal_representative_type: LegalRepresentativeType.PERSON,
+      power_of_attorney_confirmed_at: moment().toISOString(),
```
Comment (lera) [inline src/routes/business/legalRepresentative.ts]: this field is set from response

**Review by lera — APPROVED** [review]

**Pull request merged**